grin/test/AbstractInterpretation/ExtendedSyntax/OptimiseAbstractProgramSpec.hs

```diff
@@ -41,42 +40,92 @@ spec = do
       (_absInstructions absProgram1) `shouldBe` (_absInstructions absProgram2)
 
 testProgram :: Exp
-testProgram = withPrimPrelude . convertToNew $ [Old.prog|
-    grinMain = t1 <- store (CInt 1)
-               t2 <- store (CInt 10000)
-               t3 <- store (Fupto t1 t2)
-               t4 <- store (Fsum t3)
-               (CInt r') <- eval t4
-               _prim_int_print r'
+testProgram = withPrimPrelude [prog|
+  grinMain =
+    y.0 <- pure 1
+    v.0 <- pure (CInt y.0)
+    t1 <- store v.0
+    y.1 <- pure 10000
+    v.1 <- pure (CInt y.1)
+    t2 <- store v.1
+    y.3 <- pure t2
```

Comment:
Do not rename variables. Remove `var2 <- pure var1` lines. Apply copy propagation pass.

Reply:
Done